background/redux-slices/tests/nfts_update.unit.test.ts

```diff
@@ -67,11 +105,278 @@ describe("NFTs redux slice", () => {
         },
       })
 
-      const updated =
-        state.nfts[ETHEREUM.chainID][OWNER_MOCK][COLLECTION_MOCK.id]
+      const updated = extractCollection(state)
+
+      expect(updated?.name).toBe("new")
+      expect(updated?.floorPrice?.value).toEqual(2)
+    })
+  })
+
+  describe("deleteTransferred util", () => {
+    let state: NFTsSliceState
+
+    beforeEach(() => {
+      state = {
+        isReloading: false,
+        nfts: {},
+        filters: { collections: [], accounts: [], type: "desc" },
+      }
+    })
+
+    describe("when collection has missing data", () => {
+      it("should handle transfer even if collection is not saved in state", () => {
+        expect(() => deleteTransferred(state, [TRANSFER_MOCK])).not.toThrow()
+      })
+
+      it("should handle transfer if collection has many owned NFTs but none cached", () => {
+        state = initializeCollections([{ ...COLLECTION_MOCK, nftCount: 10 }])
+        deleteTransferred(state, [TRANSFER_MOCK])
+
+        const collection = extractCollection(state)
+        expect(collection).toBeDefined()
+        expect(collection?.nfts.length).toEqual(0)
+        expect(collection?.nftCount).toEqual(9)
+      })
```

Comment:
It seems to me that we should focus on unit testing.  I mean that I test a specific function and compare it with the expected state.

```suggestion
      it("should handle transfer if collection has many owned NFTs but none cached", () => {
        const newState = reducer(state, deleteTransferredNFTs([TRANSFER_MOCK]))

        const collection = extractCollection(newState)
        expect(collection).toBeDefined()
        expect(collection?.nfts.length).toEqual(0)
        expect(collection?.nftCount).toEqual(9)
      })
```

Reply:
yeah, so it is more like an integration test at this point but as we don't have the best test coverage I think it is fine if this is testing more than one function but the whole flow of saving collections and then removing them. Maybe we should move these to `_.integration.test.ts` to make it more clear?